src/components/big-interactive-pages/editor.tsx

```diff
@@ -522,236 +522,240 @@ export default function Editor({ persistenceState, cookies, roomState }: EditorP
-									);
-								}}
+								style={ outputArea.current ? {
+								height: canvasScreenSize.value.height,
+								maxHeight:  canvasScreenSize.value.maxHeight,
+									width: (1.25 * canvasScreenSize.value.height),
+									maxWidth: "100%",
+								}: { } }
+								ref={screen}
+								tabIndex={0}
+								width="1000"
+								height="800"
 							/>
-							<div
-								class={styles.helpContainer}
-								// style={{ height: realHelpAreaSize.value, maxHeight: outputArea.current?.clientHeight! - (screenControls.current?.clientHeight! + screenContainer.current?.clientHeight!) }}
-								style={{ height: realHelpAreaSize.value, maxHeight: outputArea.current?.clientHeight! - (screenContainer.current?.clientHeight! - screenControls.current?.clientHeight!) }}
+						</div>
+						<div ref={screenControls} class={styles.screenControls}>
+							<button
+								className={styles.mute}
+								onClick={() => (muted.value = !muted.value)}
+							>
+								{muted.value ? (
+									<>
+										<IoVolumeMuteOutline />{" "}
+										<span>Unmute</span>
+									</>
+								) : (
+									<>
+										<IoVolumeHighOutline />{" "}
+										<span>Mute</span>
+									</>
+								)}
+							</button>
+							<button
+								className={styles.stop}
+								onClick={onStop}
 							>
-								{!(
-									(persistenceState.value.kind === PersistenceStateKind.SHARED ||
-										persistenceState.value.kind ===
-											PersistenceStateKind.PERSISTED) &&
-									persistenceState.value.tutorial
-								) && (
+								<IoStopCircleOutline />
+								<span>Stop</span>
+							</button>
+							<div class={styles.screenSize}>
+								(Sprig screen is 1/8" / 160&times;128 px)
+							</div>
+						</div>
+					</div>
+					<div class={styles.helpContainer}>
+						<div
+							class={`${styles.horizontalResizeBar} ${
+								horizontalResizeState.value
+									? styles.resizing
+									: ""
+							}`}
+							onMouseDown={(event) => {
+								document.documentElement.style.cursor =
+									"col-resize";
+								horizontalResizeState.value = {
+									startMousePos: event.clientY,
+									startValue: realHelpAreaSize.value,
+								};
+								window.addEventListener(
+									"mouseup",
+									() => {
+										horizontalResizeState.value = null;
+										document.documentElement.style.cursor =
+											"";
+									},
+									{ once: true }
+								);
+							}}
+						/>
+						<div
+							class={styles.helpContainer}
+							// style={{ height: realHelpAreaSize.value, maxHeight: outputArea.current?.clientHeight! - (screenControls.current?.clientHeight! + screenContainer.current?.clientHeight!) }}
+							style={{ height: realHelpAreaSize.value, maxHeight: outputArea.current?.clientHeight! - (screenContainer.current?.clientHeight! - screenControls.current?.clientHeight!) }}
+						>
+							{!(
+								(persistenceState.value.kind === PersistenceStateKind.SHARED ||
```

Comment:
It really does seem like there are a multitude of these checks which probably could be collapsed into one or several utility methods to make the code more readable and less prone to making a copy/paste or logic error.

Reply:
okay but actually plz do follow up on it

Reply:
🫡